ms/MSOper/MSMetaData.cc

```diff
@@ -3878,17 +3878,23 @@ std::pair<MDirection, MDirection> MSMetaData::getPointingDirection(
 		row >= this->nRows(),
 		"Row number exceeds number of rows in the MS"
 	);
-	SHARED_PTR<Vector<Int> > ant1, ant2;
-	_getAntennas(ant1, ant2);
-	antenna1 = (*ant1)[row];
-	antenna2 = (*ant2)[row];
-	time = (*_getTimes())[row];
+	// KS note: 2015/10/30
+	// getColum method (also used in _getAntennas and _getTimes) is
+	// too expesive for the current purpose of getting single cell.
+	String ant1ColName = MeasurementSet::columnName(MSMainEnums::ANTENNA1);
+	String ant2ColName = MeasurementSet::columnName(MSMainEnums::ANTENNA2);
```

Comment:
yes this is what I meant. Indeed these copies appear all over the place in this file (though sometimes they are static), though as long as they don't show up as relevant in a performance profile its probably not necessary to change them all. I'll leave it up to you whether you consider it worthwhile to do.